src/SourceTerms/Deleptonization/CPU_Src_Deleptonization.cpp

```diff
@@ -124,9 +145,138 @@ static void Src_Deleptonization( real fluid[], const real B[],
    if ( AuxArray_Int == NULL )   printf( "ERROR : AuxArray_Int == NULL in %s !!\n", __FUNCTION__ );
 #  endif
 
-// TBF
-// profiles are stored in SrcTerms->Dlep_Profile_DataDevPtr/Dlep_Profile_RadiusDevPtr/Dlep_Profile_NBin
-// --> see "include/SrcTerms.h"
+   const double Kelvin2MeV      = Const_kB_eV*1.0e-6;
+   const real Delep_minDens_CGS = 1.0e6; // [g/cm^3]
+
+   const real Dens2CGS     = AuxArray_Flt[SRC_AUX_DENS2CGS  ];
+   const real DELEP_ENU    = AuxArray_Flt[SRC_AUX_DELEP_ENU ];
+   const real DELEP_RHO1   = AuxArray_Flt[SRC_AUX_DELEP_RHO1];
+   const real DELEP_RHO2   = AuxArray_Flt[SRC_AUX_DELEP_RHO2];
+   const real DELEP_YE1    = AuxArray_Flt[SRC_AUX_DELEP_YE1 ];
+   const real DELEP_YE2    = AuxArray_Flt[SRC_AUX_DELEP_YE2 ];
+   const real DELEP_YEC    = AuxArray_Flt[SRC_AUX_DELEP_YEC ];
+
+#  ifdef MHD
+   const real Emag       = (real)0.5*(  SQR( B[MAGX] ) + SQR( B[MAGY] ) + SQR( B[MAGZ] )  );
+#  else
+   const real Emag       = NULL_REAL;
+#  endif
+
+
+// for entropy updates
+   real Del_Ye   = NULL_REAL;
+   real Del_Entr = NULL_REAL;
+
+// output Ye
+   real Yout = NULL_REAL;
+
+
+// Deleptonization
+   const real Dens_Code = fluid[DENS];
+   const real Dens_CGS  = Dens_Code * Dens2CGS;
+   const real Eint_Code = Hydro_Con2Eint( fluid[DENS], fluid[MOMX], fluid[MOMY], fluid[MOMZ], fluid[ENGY], true, MinEint, Emag );
+         real Entr      = NULL_REAL; // entropy in kb/baryon
+         real Ye        = fluid[YE] / fluid[DENS];
+
+   real Eint_Code_0 = Eint_Code;
+   real Entr_0      = Entr;
+   real Ye_0        = Ye;
+
+
+   if ( Dens_CGS <= Delep_minDens_CGS )
+   {
+      Del_Ye = 0.0;
+   } else
+   {
+      Yout = YeOfRhoFunc( Dens_CGS, DELEP_RHO1, DELEP_RHO2,
+                          DELEP_YE1, DELEP_YE2, DELEP_YEC );
+      Del_Ye = Yout - Ye;
+      Del_Ye = MIN( 0.0, Del_Ye ); // Deleptonization cannot increase Ye
+   }
+
+   if ( Del_Ye < 0.0 )
+   {
+//    Nuclear EoS
+#     if ( NUC_TABLE_MODE == NUC_TABLE_MODE_TEMP )
+      const int  NTarget1 = 2;
+#     else
+      const int  NTarget1 = 3;
+#     endif
+            int  In_Int1[NTarget1+1];
+            real In_Flt1[3], Out1[NTarget1+1];
+
+      In_Flt1[0] = Dens_Code;
+      In_Flt1[1] = Eint_Code;
+      In_Flt1[2] = Ye;
+
+      In_Int1[0] = NTarget1;
+      In_Int1[1] = NUC_VAR_IDX_ENTR;
+      In_Int1[2] = NUC_VAR_IDX_MUNU;
+#  if ( NUC_TABLE_MODE == NUC_TABLE_MODE_ENGY )
+      In_Int1[3] = NUC_VAR_IDX_EORT;
+#  endif
+
+
+      EoS->General_FuncPtr( NUC_MODE_ENGY, Out1, In_Flt1, In_Int1, EoS->AuxArrayDevPtr_Flt, EoS->AuxArrayDevPtr_Int, EoS->Table );
+
+                 Entr      = Out1[0];
+            real mu_nu_MeV = Out1[1];
+      const real Temp_MeV  = Out1[2] * Kelvin2MeV;
+
+#  ifdef GAMER_DEBUG
+   if ( mu_nu_MeV != mu_nu_MeV ) printf( "ERROR : Couldn't get chemical potential munu (NaN) !!\n" );
+#  endif // GAMER_DEBUG
```

Comment:
Udpate the coding style. If mu_nu is always positive, use `Hydro_CheckUnphysical` instead.

Reply:
Since `mu_nu` can be negative here I just checked if `mu_nu` is `NaN`